src/action_handle_hashtags.py

```diff
@@ -0,0 +1,143 @@
+        search_button.click.wait()
+        search_button.click.wait()
+
+        print("Open hashtag #" + hashtag)
+        search_edit_text = device(resourceId='com.instagram.android:id/action_bar_search_edit_text',
+                                  className='android.widget.EditText')
+        search_edit_text.set_text(hashtag)
+
+        search_hashtag_button = device(resourceId='com.instagram.android:id/fixed_tabbar_view',
+                            className='android.widget.FrameLayout').child(index=2)
+        search_hashtag_button.click.wait()
+
+        device.wait.idle()
+        hashtag_view = device(resourceId='com.instagram.android:id/row_hashtag_textview_tag_name',
+                               className='android.widget.TextView',
+                               text="#" + hashtag)
+
+        if not hashtag_view.exists:
+            print_timeless(COLOR_FAIL + "Cannot find hashtag #" + hashtag + ", abort." + COLOR_ENDC)
+            return False
+
+        hashtag_view.click.wait()
+
+        print("Open #" + hashtag + " recent posts")
+        hashtag_type = device(resourceId='com.instagram.android:id/tab_layout',
+                                  className='android.widget.LinearLayout').child(index=1)
+        hashtag_type.click.wait()
+
+    return True
+
+def open_photo(device,index_ran):
+        # recycler_view has a className 'androidx.recyclerview.widget.RecyclerView' on modern Android versions and
+        # 'android.view.View' on Android 5.0.1 and probably earlier versions
+        recycler_view = device(resourceId='com.instagram.android:id/recycler_view',
+                                className='androidx.recyclerview.widget.RecyclerView')
+        item_view = recycler_view.child(index=index_ran)
+        item_view.click.wait()
+
+
+def _iterate_over_posts(device, interaction, storage, on_interaction):
+    #random up to 12 because min show on screen intially is 12 the max is 20
+    posts_per_screen = None
+    interactions_count = 0
+    iterations = randint(2,12)
+    while True:
+        print("Iterate over visible hashtag posts")
+        screen_iterated_posts = 0
+
+        for x in range(2,iterations):
+
+            open_photo(device,x) 
+            profile_check = device(resourceId='com.instagram.android:id/row_feed_photo_profile_imageview',
+                                    className='android.widget.ImageView')
+            profile_check.click.wait()
+            
+            profile_name = device(resourceId='com.instagram.android:id/action_bar_textview_title',
+                                    className='android.widget.TextView')
+            username = profile_name.text
+
+            screen_iterated_posts =+ 1
+            if storage.check_user_was_interacted(username):
+                print("@" + username + ": already interacted. Skip.")
+                device.press.back()
+                device.press.back()
+            elif storage.check_user_was_interacted_recently(username):

```

Comment:
It's unnecessary to check twice here: `check_user_was_interacted_recently(username)` will always return False if `check_user_was_interacted(username)` returns False.

But are you sure that we need it here? I used `check_user_was_interacted_recently(username)` for interacting with self followers only.

Reply:
Thats true I can take that out